packages/network-controller/src/create-network-client.ts

```diff
@@ -0,0 +1,220 @@
+import {
+  createAsyncMiddleware,
+  createScaffoldMiddleware,
+  JsonRpcEngine,
+  mergeMiddleware,
+  JsonRpcMiddleware,
+} from 'json-rpc-engine';
+import {
+  createBlockCacheMiddleware,
+  createBlockRefMiddleware,
+  createBlockRefRewriteMiddleware,
+  createBlockTrackerInspectorMiddleware,
+  createInflightCacheMiddleware,
+  createFetchMiddleware,
+  createRetryOnEmptyMiddleware,
+} from '@metamask/eth-json-rpc-middleware';
+import {
+  providerFromEngine,
+  providerFromMiddleware,
+  SafeEventEmitterProvider,
+} from '@metamask/eth-json-rpc-provider';
+import { createInfuraMiddleware } from '@metamask/eth-json-rpc-infura';
+import type { Hex } from '@metamask/utils';
+import { PollingBlockTracker } from 'eth-block-tracker';
+import { InfuraNetworkType, NetworksChainId } from '@metamask/controller-utils';
+
+const SECOND = 1000;
+
+export enum NetworkClientType {
```

Comment:
I'd ideally like to add JSDocs for types, but it's a direct copy from the extension. Since these are minor updates, we can always update it later if we like.